src/data/setoid/basic.lean

```diff
@@ -359,4 +368,23 @@ def correspondence (r : setoid α) : {s // r ≤ s} ≃o setoid (quotient r) :=
       t.1.trans' (t.1.symm' $ t.2 $ eq_rel.1 hx) $ t.1.trans' ht $ t.2 $ eq_rel.1 hy,
       λ h x y hs, let ⟨a, b, hx, hy, Hs⟩ := hs in ⟨a, b, hx, hy, h Hs⟩⟩ }
 
+@[simp]
+lemma subsingleton_quotient_iff_eq_top  {s : setoid α} :
```

Comment:
I think a better name for this might be `quotient.subsingleton_iff` and similarly with `quot.subsingleton_iff` below.
You'll want to move these out of the `setoid` namespace too.

Reply:
Done